Enterprise-CMCS/cmcs-eregulations
# EREGCSC- 2875 subject counts -- Implement counts of filtered search results by subject

Resolves [EREGCSC-2875](https://jiraent.cms.gov/browse/EREGCSC-2875)

**Description**

This PR will unhide the Subjects Dropdown on the Search page and add counts to the list items in the Subjects Dropdown.

**This pull request changes:**

- Refactor `fetchCategoriesContainer` into `FetchItemsContainer` to be more generic how it can be used
   - It is now used to fetch Categories and Subjects
- Add ability for `FetchItemsContainer` to also fetch the counts of the type of items it is fetching
- Add logic in `dropdowns/Subjects.vue` component to transform fetched Subjects and fetched Counts into an abbreviated list of Subjects that are associated with at least one document
   - This list retains the sort order of the counts response, which is sorted descending by subject count
- Updates counts logic to ensure that counts only change when the search query changes
   - Counts in Document Type Selector will not change if Categories or Subjects are selected
   - Counts in Subjects dropdown will not change if Document Type is changed or if Categories are selected
- remove loading indicator from Categories dropdown to make it consistent with the Subjects dropdown, which being a different Vuetify component has a different loading indicator.  Both will still be disabled while fetching from the server and will have a consistent grayed-out disabled style
- Add Vitest unit tests and Cypress end to end tests where appropriate

**Steps to manually verify this change:**

1. Green check marks
2. Visit [Experimental deployment](https://tbrtjxp577.execute-api.us-east-1.amazonaws.com/dev1452/search/?q=test) and ensure that Subjects dropdown is present when searching
3. Ensure that both Subjects and Categories dropdowns are hidden when only Regulations is checked
4. Ensure that both a category and a subject can be selected at the same time
5. Ensure that counts are properly displayed in the Subjects dropdown and match the [counts API call
](https://tbrtjxp577.execute-api.us-east-1.amazonaws.com/dev1452/v3/content-search/counts?q=test)
